File: camerax_android/lib/src/legacy/common/range.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension IntRangeArgs on $interface.Range<int> {
  $native.IntRange get obj {
    return $native.IntRange(
      lower: lower,
      upper: upper,
    );
  }
}

extension IntRangeObj on $native.IntRange {
  $interface.Range<int> get args {
    return $interface.Range(lower, upper);
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/common/Range.kt
package dev.hebei.camerax_android.legacy.common

import android.util.Range
import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiIntRange

class IntRange(registrar: CameraXRegistrar) : PigeonApiIntRange(registrar) {
    override fun pigeon_defaultConstructor(lower: Long, upper: Long): Stub {
        val obj = Range(lower.toInt(), upper.toInt())
        return Stub(obj)
    }

    override fun lower(pigeon_instance: Stub): Long {
        return pigeon_instance.lower
    }

    override fun upper(pigeon_instance: Stub): Long {
        return pigeon_instance.upper
    }

    class Stub(internal val obj: Range<Int>) {
        val lower get() = obj.lower.toLong()
        val upper get() = obj.upper.toLong()
    }
}